webapp/src/test/java/com/google/sps/servlets/RunValidatorTest.java

```diff
@@ -0,0 +1,55 @@
+// //
+// //     https://www.apache.org/licenses/LICENSE-2.0
+// //
+// // Unless required by applicable law or agreed to in writing, software
+// // distributed under the License is distributed on an "AS IS" BASIS,
+// // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// // See the License for the specific language governing permissions and
+// // limitations under the License.
+package com.google.sps.servlets;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.lang.ClassLoader;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
+import org.apache.commons.fileupload.FileItem;
+import org.apache.commons.fileupload.disk.DiskFileItemFactory;
+import org.apache.commons.fileupload.servlet.ServletFileUpload;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mobilitydata.gtfsvalidator.input.GtfsFeedName;
+import org.mobilitydata.gtfsvalidator.input.GtfsInput;
+import org.mobilitydata.gtfsvalidator.notice.NoticeContainer;
+import org.mobilitydata.gtfsvalidator.table.GtfsFeedContainer;
+import org.mobilitydata.gtfsvalidator.table.GtfsFeedLoader;
+import org.mobilitydata.gtfsvalidator.validator.ValidatorLoader;
+
+@RunWith(JUnit4.class)
+public class RunValidatorTest {
+  @Test
+  public void testSimpleGtfsFeed() {
+    String json = UploadServlet.runValidator("src/test/resources/SAMPLE.zip", 1, "au-sydney-buses");
+    String expected =
+        "{\"notices\":[{\"code\":\"unknown_column\",\"totalNotices\":1,\"notices\":[{\"filename\":\"stop_times.txt\",\"fieldName\":\"drop_off_time\",\"index\":8}]},{\"code\":\"invalid_row_length\",\"totalNotices\":13,\"notices\":[{\"filename\":\"stop_times.txt\",\"csvRowNumber\":17,\"rowLength\":5,\"headerCount\":9},{\"filename\":\"stop_times.txt\",\"csvRowNumber\":18,\"rowLength\":5,\"headerCount\":9},{\"filename\":\"stop_times.txt\",\"csvRowNumber\":19,\"rowLength\":5,\"headerCount\":9},{\"filename\":\"stop_times.txt\",\"csvRowNumber\":20,\"rowLength\":5,\"headerCount\":9},{\"filename\":\"stop_times.txt\",\"csvRowNumber\":21,\"rowLength\":5,\"headerCount\":9},{\"filename\":\"stop_times.txt\",\"csvRowNumber\":22,\"rowLength\":5,\"headerCount\":9},{\"filename\":\"stop_times.txt\",\"csvRowNumber\":23,\"rowLength\":5,\"headerCount\":9},{\"filename\":\"stop_times.txt\",\"csvRowNumber\":24,\"rowLength\":5,\"headerCount\":9},{\"filename\":\"stop_times.txt\",\"csvRowNumber\":25,\"rowLength\":5,\"headerCount\":9},{\"filename\":\"stop_times.txt\",\"csvRowNumber\":26,\"rowLength\":5,\"headerCount\":9},{\"filename\":\"stop_times.txt\",\"csvRowNumber\":27,\"rowLength\":5,\"headerCount\":9},{\"filename\":\"stop_times.txt\",\"csvRowNumber\":28,\"rowLength\":5,\"headerCount\":9},{\"filename\":\"stop_times.txt\",\"csvRowNumber\":29,\"rowLength\":5,\"headerCount\":9}]}]}";
+    Assert.assertEquals(json, expected);
```

Comment:
https://testing.googleblog.com/2015/01/testing-on-toilet-change-detector-tests.html

Can you possibly think of a more generic expectations that would fail if the verifier executed at all but would not be dependent on specific error detection?

Reply:
I am now just checking that it identifies the error e.g. that this is contained in the notice "\"code\":\"invalid_row_length\",\"totalNotices\":13" with google's truth library